packages/go_router/lib/src/configuration.dart

```diff
@@ -111,6 +130,108 @@ class RouteConfiguration {
     return true;
   }
 
+  // Check to see that the configured initialLocation of StackedShellBranches
+  // points to a descendant route of the route branch.
+  bool _debugCheckStackedShellBranchDefaultLocations(
+      List<RouteBase> routes, RouteMatcher matcher) {
+    try {
+      for (final RouteBase route in routes) {
+        if (route is StackedShellRoute) {
+          for (final StackedShellBranch branch in route.branches) {
+            if (branch.initialLocation == null) {
+              // Recursively search for the first GoRoute descendant. Will
+              // throw assertion error if not found.
+              findStackedShellBranchDefaultLocation(branch);
+            } else {
+              final RouteBase initialLocationRoute =
+                  matcher.findMatch(branch.initialLocation!).last.route;
+              final RouteBase? match = branch.routes.firstWhereOrNull(
+                  (RouteBase e) => _debugIsDescendantOrSame(
+                      ancestor: e, route: initialLocationRoute));
+              assert(
+                  match != null,
+                  'The initialLocation (${branch.initialLocation}) of '
+                  'StackedShellBranch must match a descendant route of the '
+                  'branch');
+            }
+          }
+        }
+        _debugCheckStackedShellBranchDefaultLocations(route.routes, matcher);
+      }
+    } on MatcherError catch (e) {
+      assert(
+          false,
+          'initialLocation (${e.location}) of StackedShellBranch must '
+          'be a valid location');
+    }
+    return true;
+  }
+
+  /// Returns an Iterable that traverses the provided routes and their
+  /// sub-routes recursively.
+  static Iterable<RouteBase> routesRecursively(
+      Iterable<RouteBase> routes) sync* {
+    for (final RouteBase route in routes) {
+      yield route;
+      yield* routesRecursively(route.routes);
+    }
+  }
+
+  static GoRoute? _findFirstGoRoute(List<RouteBase> routes) =>
+      routesRecursively(routes).whereType<GoRoute>().firstOrNull;
+
+  /// Tests if a route is a descendant of, or same as, an ancestor route.
+  bool _debugIsDescendantOrSame(
+          {required RouteBase ancestor, required RouteBase route}) =>
+      ancestor == route || routesRecursively(ancestor.routes).contains(route);
+
+  /// Recursively traverses the routes of the provided StackedShellBranch to
+  /// find the first GoRoute, from which a full path will be derived.
+  String findStackedShellBranchDefaultLocation(StackedShellBranch branch) {
```

Comment:
this can be static

Reply:
I still think this should be in the route.dart
you can do that if _fullPathForRoute is a public method and should probably be put into path_utils

Reply:
Sounds good 👍

Reply:
On second thought, this is unfortunately not possible, since the `routes` field of RouteConfiguration is needed in findStackedShellBranchDefaultLocation.

Reply:
I think I managed to resolve this by adding a `locationForRoute` method to `GoRouter` (and `RouteConfiguration`).